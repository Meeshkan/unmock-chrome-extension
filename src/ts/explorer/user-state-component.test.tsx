import * as React from "react";
import UserStateComponent from "./user-state-component";
import { shallow } from "enzyme";

const mockUserState = {
  actions: [{ type: "log" }],
  activities: {},
  changed: true,
  context: { path: [] },
  event: { type: "NEXT" },
  events: [],
  history: {
    actions: [{ type: "log" }],
    activities: {},
    context: { path: [] },
    event: { type: "NEXT" },
    events: [],
    historyValue: { current: "addOperation", states: {} },
    meta: {},
    value: "addOperation",
  },
  historyValue: { current: "addPath", states: {} },
  meta: {},
  value: "addPath",
};

describe("User state component", () => {
  test("renders correctly", () => {
    const wrapper = shallow(<UserStateComponent userState={mockUserState} />);
    expect(wrapper).toMatchSnapshot();
  });
});
