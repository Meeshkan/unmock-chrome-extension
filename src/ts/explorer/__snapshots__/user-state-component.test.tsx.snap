// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User state component renders correctly 1`] = `
<div>
  <div
    id="active"
  >
    <h3>
      User state
    </h3>
    <Alert
      closeLabel="Close alert"
      show={true}
      transition={[Function]}
      variant="info"
    >
      Active user state: 
      {
  "actions": [
    {
      "type": "log"
    }
  ],
  "activities": {},
  "changed": true,
  "context": {
    "path": []
  },
  "event": {
    "type": "NEXT"
  },
  "events": [],
  "history": {
    "actions": [
      {
        "type": "log"
      }
    ],
    "activities": {},
    "context": {
      "path": []
    },
    "event": {
      "type": "NEXT"
    },
    "events": [],
    "historyValue": {
      "current": "addOperation",
      "states": {}
    },
    "meta": {},
    "value": "addOperation"
  },
  "historyValue": {
    "current": "addPath",
    "states": {}
  },
  "meta": {},
  "value": "addPath"
}
    </Alert>
  </div>
</div>
`;
